### PR TITLE
CloneDbConnection handles AccessTokenCallback

### PR DESCRIPTION
**Pull request by rayao**
when cloning, properly copy AccessTokenCallback

### CODE REVIEW

File: src/ErikEJ.EntityFramework.SqlServer/MicrosoftSqlProviderServices.cs [hunk 1]
```diff
@@ -1540,6 +1540,16 @@ public override DbConnection CloneDbConnection(DbConnection connection, DbProvid
             DebugCheck.NotNull(connection);
             DebugCheck.NotNull(factory);
 
+            if (connection is SqlConnection sqlConnection && sqlConnection.AccessTokenCallback != null)
+            {
+                // SqlConnection.Clone doesn't handle this case, so we special-case it here, if later it's fixed in
+                // Microsoft.Data.SqlCLient, we can remove this code piece.
```
Comment (ErikEJ) [inline src/ErikEJ.EntityFramework.SqlServer/MicrosoftSqlProviderServices.cs:1546]: Pls add a link to the SqlClient repo related issue

File: src/ErikEJ.EntityFramework.SqlServer/ErikEJ.EntityFramework.SqlServer.csproj [hunk 2]
```diff
@@ -113,7 +113,7 @@
   <ItemGroup>
 	<PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1" PrivateAssets="All" />	  
     <PackageReference Include="EntityFramework" Version="6.4.4" />
-    <PackageReference Include="Microsoft.Data.SqlClient" Version="5.1.5" />
+    <PackageReference Include="Microsoft.Data.SqlClient" Version="5.2.0" />
```
Comment (ErikEJ) [inline src/ErikEJ.EntityFramework.SqlServer/ErikEJ.EntityFramework.SqlServer.csproj:116]: Can we avoid this for now, I am waiting for Bug fixes in 5.2.1

**Comment by ErikEJ** [comment]
Maybe add a little something to the readme as well?

File: src/ErikEJ.EntityFramework.SqlServer/ErikEJ.EntityFramework.SqlServer.csproj (hunk 2 above)
Comment (ErikEJ) [reply src/ErikEJ.EntityFramework.SqlServer/ErikEJ.EntityFramework.SqlServer.csproj:116]: Ah!

**Comment by rayao** [comment]
Thank you, Erik!

**Comment by ErikEJ** [comment]
@rayao Have you been able to verify that the updated package works as expected?

**Comment by rayao** [comment]
@ErikEJ Our CI system needs some time to push a change to deployment, I'll report later today.

**Comment by rayao** [comment]
It's just been proved that the updated package works as expected, we're unblocked.
Thanks very much! @ErikEJ

**Comment by ErikEJ** [comment]
Fantastic, thanks for confirming!

**Pull request merged**